### PR TITLE
Create mergeConfig in Lavamoat-Core

### PR DESCRIPTION
**Pull request by EtDu**
- Creates new `mergeConfig` that uses `merge-deep` dep to recursively merge full config objects
- Add test for new `mergeConfig`

### CODE REVIEW

File: packages/tofu/src/util.js [hunk 1]
```diff
@@ -122,6 +124,14 @@ function mergeConfig (configA, configB) {
   return newConfig
 }
 
+function mergeConfig (configA, configB) {
+  configA = JSON.parse(configA)
```
Comment (kumavis) [inline packages/tofu/src/util.js]: why are we calling parse here? that wasn't part of the original fn. plz remove

File: packages/tofu/test/util.js [hunk 2]
```diff
@@ -29,10 +42,48 @@ testMerge('non-overlapping', {
   'xyz.jkl': 'write'
 })
 
-function testMerge (label, configA, configB, expectedResultObj) {
+testMerge('merge with resources', {
```
Comment (kumavis) [inline packages/tofu/test/util.js]: 👍

**Review by kumavis — APPROVED** [review]

**Review by kumavis — CHANGES_REQUESTED** [review]
meant to mark as request changes

File: packages/tofu/src/util.js (hunk 1 above)
Comment (EtDu) [reply packages/tofu/src/util.js]: because we're passing in the string from readfile. Unless we want to use parse on the consumer side

Comment (kumavis) [reply packages/tofu/src/util.js]: parse on consumer side

File: yarn.lock [hunk 3]
```diff
@@ -9999,6 +9999,19 @@ yargs@^15.0.2, yargs@^15.3.1:
     y18n "^4.0.0"
     yargs-parser "^18.1.1"
 
+yargs@^16.0.0:
```
Comment (kumavis) [inline yarn.lock:10002]: weird

**Review by kumavis — APPROVED** [review]
approve

**Pull request merged**